Argument highlighter improvement

TRMMax opened the pull request:
Fixed text arguments not being recognised as block arguments.

TRMMax committed:
argument highlighter fix

Fixed text arguments not being recognised as block arguments.

homchom commented:
This should probably be improved anyway. There could be an enum type that maps both ids or something, so the switch statement could be removed entirely.

TRMMax committed:
Replaced argument error checking system

Replaced the large switch statement with a dedicated enum type with all possible value types for an argument.

TRMMax committed:
small argument type compatibility fix
